// @aws-sdk/client-s3/dist/es/models/index.js
import { isa as __isa } from "@aws-sdk/smithy-client";
export var AbortIncompleteMultipartUpload;
(function (AbortIncompleteMultipartUpload) {
    AbortIncompleteMultipartUpload.isa = function (o) {
        return __isa(o, "AbortIncompleteMultipartUpload");
    };
})(AbortIncompleteMultipartUpload || (AbortIncompleteMultipartUpload = {}));
export var AbortMultipartUploadOutput;
(function (AbortMultipartUploadOutput) {
    AbortMultipartUploadOutput.isa = function (o) {
        return __isa(o, "AbortMultipartUploadOutput");
    };
})(AbortMultipartUploadOutput || (AbortMultipartUploadOutput = {}));
export var AbortMultipartUploadRequest;
(function (AbortMultipartUploadRequest) {
    AbortMultipartUploadRequest.isa = function (o) {
        return __isa(o, "AbortMultipartUploadRequest");
    };
})(AbortMultipartUploadRequest || (AbortMultipartUploadRequest = {}));
export var AccelerateConfiguration;
(function (AccelerateConfiguration) {
    AccelerateConfiguration.isa = function (o) {
        return __isa(o, "AccelerateConfiguration");
    };
})(AccelerateConfiguration || (AccelerateConfiguration = {}));
export var AccessControlPolicy;
(function (AccessControlPolicy) {
    AccessControlPolicy.isa = function (o) {
        return __isa(o, "AccessControlPolicy");
    };
})(AccessControlPolicy || (AccessControlPolicy = {}));
export var AccessControlTranslation;
(function (AccessControlTranslation) {
    AccessControlTranslation.isa = function (o) {
        return __isa(o, "AccessControlTranslation");
    };
})(AccessControlTranslation || (AccessControlTranslation = {}));
export var AnalyticsAndOperator;
(function (AnalyticsAndOperator) {
    AnalyticsAndOperator.isa = function (o) {
        return __isa(o, "AnalyticsAndOperator");
    };
})(AnalyticsAndOperator || (AnalyticsAndOperator = {}));
export var AnalyticsConfiguration;
(function (AnalyticsConfiguration) {
    AnalyticsConfiguration.isa = function (o) {
        return __isa(o, "AnalyticsConfiguration");
    };
})(AnalyticsConfiguration || (AnalyticsConfiguration = {}));
export var AnalyticsExportDestination;
(function (AnalyticsExportDestination) {
    AnalyticsExportDestination.isa = function (o) {
        return __isa(o, "AnalyticsExportDestination");
    };
})(AnalyticsExportDestination || (AnalyticsExportDestination = {}));
export var AnalyticsFilter;
(function (AnalyticsFilter) {
    AnalyticsFilter.isa = function (o) {
        return __isa(o, "AnalyticsFilter");
    };
})(AnalyticsFilter || (AnalyticsFilter = {}));
export var AnalyticsS3BucketDestination;
(function (AnalyticsS3BucketDestination) {
    AnalyticsS3BucketDestination.isa = function (o) {
        return __isa(o, "AnalyticsS3BucketDestination");
    };
})(AnalyticsS3BucketDestination || (AnalyticsS3BucketDestination = {}));
export var Bucket;
(function (Bucket) {
    Bucket.isa = function (o) { return __isa(o, "Bucket"); };
})(Bucket || (Bucket = {}));
export var BucketAlreadyExists;
(function (BucketAlreadyExists) {
    BucketAlreadyExists.isa = function (o) {
        return __isa(o, "BucketAlreadyExists");
    };
})(BucketAlreadyExists || (BucketAlreadyExists = {}));
export var BucketAlreadyOwnedByYou;
(function (BucketAlreadyOwnedByYou) {
    BucketAlreadyOwnedByYou.isa = function (o) {
        return __isa(o, "BucketAlreadyOwnedByYou");
    };
})(BucketAlreadyOwnedByYou || (BucketAlreadyOwnedByYou = {}));
export var BucketLifecycleConfiguration;
(function (BucketLifecycleConfiguration) {
    BucketLifecycleConfiguration.isa = function (o) {
        return __isa(o, "BucketLifecycleConfiguration");
    };
})(BucketLifecycleConfiguration || (BucketLifecycleConfiguration = {}));
export var BucketLoggingStatus;
(function (BucketLoggingStatus) {
    BucketLoggingStatus.isa = function (o) {
        return __isa(o, "BucketLoggingStatus");
    };
})(BucketLoggingStatus || (BucketLoggingStatus = {}));
export var CORSConfiguration;
(function (CORSConfiguration) {
    CORSConfiguration.isa = function (o) {
        return __isa(o, "CORSConfiguration");
    };
})(CORSConfiguration || (CORSConfiguration = {}));
export var CORSRule;
(function (CORSRule) {
    CORSRule.isa = function (o) { return __isa(o, "CORSRule"); };
})(CORSRule || (CORSRule = {}));
export var CSVInput;
(function (CSVInput) {
    CSVInput.isa = function (o) { return __isa(o, "CSVInput"); };
})(CSVInput || (CSVInput = {}));
export var CSVOutput;
(function (CSVOutput) {
    CSVOutput.isa = function (o) { return __isa(o, "CSVOutput"); };
})(CSVOutput || (CSVOutput = {}));
export var CommonPrefix;
(function (CommonPrefix) {
    CommonPrefix.isa = function (o) { return __isa(o, "CommonPrefix"); };
})(CommonPrefix || (CommonPrefix = {}));
export var CompleteMultipartUploadOutput;
(function (CompleteMultipartUploadOutput) {
    CompleteMultipartUploadOutput.isa = function (o) {
        return __isa(o, "CompleteMultipartUploadOutput");
    };
})(CompleteMultipartUploadOutput || (CompleteMultipartUploadOutput = {}));
export var CompleteMultipartUploadRequest;
(function (CompleteMultipartUploadRequest) {
    CompleteMultipartUploadRequest.isa = function (o) {
        return __isa(o, "CompleteMultipartUploadRequest");
    };
})(CompleteMultipartUploadRequest || (CompleteMultipartUploadRequest = {}));
export var CompletedMultipartUpload;
(function (CompletedMultipartUpload) {
    CompletedMultipartUpload.isa = function (o) {
        return __isa(o, "CompletedMultipartUpload");
    };
})(CompletedMultipartUpload || (CompletedMultipartUpload = {}));
export var CompletedPart;
(function (CompletedPart) {
    CompletedPart.isa = function (o) { return __isa(o, "CompletedPart"); };
})(CompletedPart || (CompletedPart = {}));
export var Condition;
(function (Condition) {
    Condition.isa = function (o) { return __isa(o, "Condition"); };
})(Condition || (Condition = {}));
export var ContinuationEvent;
(function (ContinuationEvent) {
    ContinuationEvent.isa = function (o) {
        return __isa(o, "ContinuationEvent");
    };
})(ContinuationEvent || (ContinuationEvent = {}));
export var CopyObjectOutput;
(function (CopyObjectOutput) {
    CopyObjectOutput.isa = function (o) {
        return __isa(o, "CopyObjectOutput");
    };
})(CopyObjectOutput || (CopyObjectOutput = {}));
export var CopyObjectRequest;
(function (CopyObjectRequest) {
    CopyObjectRequest.isa = function (o) {
        return __isa(o, "CopyObjectRequest");
    };
})(CopyObjectRequest || (CopyObjectRequest = {}));
export var CopyObjectResult;
(function (CopyObjectResult) {
    CopyObjectResult.isa = function (o) {
        return __isa(o, "CopyObjectResult");
    };
})(CopyObjectResult || (CopyObjectResult = {}));
export var CopyPartResult;
(function (CopyPartResult) {
    CopyPartResult.isa = function (o) {
        return __isa(o, "CopyPartResult");
    };
})(CopyPartResult || (CopyPartResult = {}));
export var CreateBucketConfiguration;
(function (CreateBucketConfiguration) {
    CreateBucketConfiguration.isa = function (o) {
        return __isa(o, "CreateBucketConfiguration");
    };
})(CreateBucketConfiguration || (CreateBucketConfiguration = {}));
export var CreateBucketOutput;
(function (CreateBucketOutput) {
    CreateBucketOutput.isa = function (o) {
        return __isa(o, "CreateBucketOutput");
    };
})(CreateBucketOutput || (CreateBucketOutput = {}));
export var CreateBucketRequest;
(function (CreateBucketRequest) {
    CreateBucketRequest.isa = function (o) {
        return __isa(o, "CreateBucketRequest");
    };
})(CreateBucketRequest || (CreateBucketRequest = {}));
export var CreateMultipartUploadOutput;
(function (CreateMultipartUploadOutput) {
    CreateMultipartUploadOutput.isa = function (o) {
        return __isa(o, "CreateMultipartUploadOutput");
    };
})(CreateMultipartUploadOutput || (CreateMultipartUploadOutput = {}));
export var CreateMultipartUploadRequest;
(function (CreateMultipartUploadRequest) {
    CreateMultipartUploadRequest.isa = function (o) {
        return __isa(o, "CreateMultipartUploadRequest");
    };
})(CreateMultipartUploadRequest || (CreateMultipartUploadRequest = {}));
export var DefaultRetention;
(function (DefaultRetention) {
    DefaultRetention.isa = function (o) {
        return __isa(o, "DefaultRetention");
    };
})(DefaultRetention || (DefaultRetention = {}));
export var Delete;
(function (Delete) {
    Delete.isa = function (o) { return __isa(o, "Delete"); };
})(Delete || (Delete = {}));
export var DeleteBucketAnalyticsConfigurationRequest;
(function (DeleteBucketAnalyticsConfigurationRequest) {
    DeleteBucketAnalyticsConfigurationRequest.isa = function (o) {
        return __isa(o, "DeleteBucketAnalyticsConfigurationRequest");
    };
})(DeleteBucketAnalyticsConfigurationRequest || (DeleteBucketAnalyticsConfigurationRequest = {}));
export var DeleteBucketCorsRequest;
(function (DeleteBucketCorsRequest) {
    DeleteBucketCorsRequest.isa = function (o) {
        return __isa(o, "DeleteBucketCorsRequest");
    };
})(DeleteBucketCorsRequest || (DeleteBucketCorsRequest = {}));
export var DeleteBucketEncryptionRequest;
(function (DeleteBucketEncryptionRequest) {
    DeleteBucketEncryptionRequest.isa = function (o) {
        return __isa(o, "DeleteBucketEncryptionRequest");
    };
})(DeleteBucketEncryptionRequest || (DeleteBucketEncryptionRequest = {}));
export var DeleteBucketInventoryConfigurationRequest;
(function (DeleteBucketInventoryConfigurationRequest) {
    DeleteBucketInventoryConfigurationRequest.isa = function (o) {
        return __isa(o, "DeleteBucketInventoryConfigurationRequest");
    };
})(DeleteBucketInventoryConfigurationRequest || (DeleteBucketInventoryConfigurationRequest = {}));
export var DeleteBucketLifecycleRequest;
(function (DeleteBucketLifecycleRequest) {
    DeleteBucketLifecycleRequest.isa = function (o) {
        return __isa(o, "DeleteBucketLifecycleRequest");
    };
})(DeleteBucketLifecycleRequest || (DeleteBucketLifecycleRequest = {}));
export var DeleteBucketMetricsConfigurationRequest;
(function (DeleteBucketMetricsConfigurationRequest) {
    DeleteBucketMetricsConfigurationRequest.isa = function (o) {
        return __isa(o, "DeleteBucketMetricsConfigurationRequest");
    };
})(DeleteBucketMetricsConfigurationRequest || (DeleteBucketMetricsConfigurationRequest = {}));
export var DeleteBucketPolicyRequest;
(function (DeleteBucketPolicyRequest) {
    DeleteBucketPolicyRequest.isa = function (o) {
        return __isa(o, "DeleteBucketPolicyRequest");
    };
})(DeleteBucketPolicyRequest || (DeleteBucketPolicyRequest = {}));
export var DeleteBucketReplicationRequest;
(function (DeleteBucketReplicationRequest) {
    DeleteBucketReplicationRequest.isa = function (o) {
        return __isa(o, "DeleteBucketReplicationRequest");
    };
})(DeleteBucketReplicationRequest || (DeleteBucketReplicationRequest = {}));
export var DeleteBucketRequest;
(function (DeleteBucketRequest) {
    DeleteBucketRequest.isa = function (o) {
        return __isa(o, "DeleteBucketRequest");
    };
})(DeleteBucketRequest || (DeleteBucketRequest = {}));
export var DeleteBucketTaggingRequest;
(function (DeleteBucketTaggingRequest) {
    DeleteBucketTaggingRequest.isa = function (o) {
        return __isa(o, "DeleteBucketTaggingRequest");
    };
})(DeleteBucketTaggingRequest || (DeleteBucketTaggingRequest = {}));
export var DeleteBucketWebsiteRequest;
(function (DeleteBucketWebsiteRequest) {
    DeleteBucketWebsiteRequest.isa = function (o) {
        return __isa(o, "DeleteBucketWebsiteRequest");
    };
})(DeleteBucketWebsiteRequest || (DeleteBucketWebsiteRequest = {}));
export var DeleteMarkerEntry;
(function (DeleteMarkerEntry) {
    DeleteMarkerEntry.isa = function (o) {
        return __isa(o, "DeleteMarkerEntry");
    };
})(DeleteMarkerEntry || (DeleteMarkerEntry = {}));
export var DeleteMarkerReplication;
(function (DeleteMarkerReplication) {
    DeleteMarkerReplication.isa = function (o) {
        return __isa(o, "DeleteMarkerReplication");
    };
})(DeleteMarkerReplication || (DeleteMarkerReplication = {}));
export var DeleteObjectOutput;
(function (DeleteObjectOutput) {
    DeleteObjectOutput.isa = function (o) {
        return __isa(o, "DeleteObjectOutput");
    };
})(DeleteObjectOutput || (DeleteObjectOutput = {}));
export var DeleteObjectRequest;
(function (DeleteObjectRequest) {
    DeleteObjectRequest.isa = function (o) {
        return __isa(o, "DeleteObjectRequest");
    };
})(DeleteObjectRequest || (DeleteObjectRequest = {}));
export var DeleteObjectTaggingOutput;
(function (DeleteObjectTaggingOutput) {
    DeleteObjectTaggingOutput.isa = function (o) {
        return __isa(o, "DeleteObjectTaggingOutput");
    };
})(DeleteObjectTaggingOutput || (DeleteObjectTaggingOutput = {}));
export var DeleteObjectTaggingRequest;
(function (DeleteObjectTaggingRequest) {
    DeleteObjectTaggingRequest.isa = function (o) {
        return __isa(o, "DeleteObjectTaggingRequest");
    };
})(DeleteObjectTaggingRequest || (DeleteObjectTaggingRequest = {}));
export var DeleteObjectsOutput;
(function (DeleteObjectsOutput) {
    DeleteObjectsOutput.isa = function (o) {
        return __isa(o, "DeleteObjectsOutput");
    };
})(DeleteObjectsOutput || (DeleteObjectsOutput = {}));
export var DeleteObjectsRequest;
(function (DeleteObjectsRequest) {
    DeleteObjectsRequest.isa = function (o) {
        return __isa(o, "DeleteObjectsRequest");
    };
})(DeleteObjectsRequest || (DeleteObjectsRequest = {}));
export var DeletePublicAccessBlockRequest;
(function (DeletePublicAccessBlockRequest) {
    DeletePublicAccessBlockRequest.isa = function (o) {
        return __isa(o, "DeletePublicAccessBlockRequest");
    };
})(DeletePublicAccessBlockRequest || (DeletePublicAccessBlockRequest = {}));
export var DeletedObject;
(function (DeletedObject) {
    DeletedObject.isa = function (o) { return __isa(o, "DeletedObject"); };
})(DeletedObject || (DeletedObject = {}));
export var Destination;
(function (Destination) {
    Destination.isa = function (o) { return __isa(o, "Destination"); };
})(Destination || (Destination = {}));
export var Encryption;
(function (Encryption) {
    Encryption.isa = function (o) { return __isa(o, "Encryption"); };
})(Encryption || (Encryption = {}));
export var EncryptionConfiguration;
(function (EncryptionConfiguration) {
    EncryptionConfiguration.isa = function (o) {
        return __isa(o, "EncryptionConfiguration");
    };
})(EncryptionConfiguration || (EncryptionConfiguration = {}));
export var EndEvent;
(function (EndEvent) {
    EndEvent.isa = function (o) { return __isa(o, "EndEvent"); };
})(EndEvent || (EndEvent = {}));
export var _Error;
(function (_Error) {
    _Error.isa = function (o) { return __isa(o, "Error"); };
})(_Error || (_Error = {}));
export var ErrorDocument;
(function (ErrorDocument) {
    ErrorDocument.isa = function (o) { return __isa(o, "ErrorDocument"); };
})(ErrorDocument || (ErrorDocument = {}));
export var ExistingObjectReplication;
(function (ExistingObjectReplication) {
    ExistingObjectReplication.isa = function (o) {
        return __isa(o, "ExistingObjectReplication");
    };
})(ExistingObjectReplication || (ExistingObjectReplication = {}));
export var ExpressionType;
(function (ExpressionType) {
    ExpressionType["SQL"] = "SQL";
})(ExpressionType || (ExpressionType = {}));
export var FileHeaderInfo;
(function (FileHeaderInfo) {
    FileHeaderInfo["IGNORE"] = "IGNORE";
    FileHeaderInfo["NONE"] = "NONE";
    FileHeaderInfo["USE"] = "USE";
})(FileHeaderInfo || (FileHeaderInfo = {}));
export var FilterRule;
(function (FilterRule) {
    FilterRule.isa = function (o) { return __isa(o, "FilterRule"); };
})(FilterRule || (FilterRule = {}));
export var GetBucketAccelerateConfigurationOutput;
(function (GetBucketAccelerateConfigurationOutput) {
    GetBucketAccelerateConfigurationOutput.isa = function (o) {
        return __isa(o, "GetBucketAccelerateConfigurationOutput");
    };
})(GetBucketAccelerateConfigurationOutput || (GetBucketAccelerateConfigurationOutput = {}));
export var GetBucketAccelerateConfigurationRequest;
(function (GetBucketAccelerateConfigurationRequest) {
    GetBucketAccelerateConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketAccelerateConfigurationRequest");
    };
})(GetBucketAccelerateConfigurationRequest || (GetBucketAccelerateConfigurationRequest = {}));
export var GetBucketAclOutput;
(function (GetBucketAclOutput) {
    GetBucketAclOutput.isa = function (o) {
        return __isa(o, "GetBucketAclOutput");
    };
})(GetBucketAclOutput || (GetBucketAclOutput = {}));
export var GetBucketAclRequest;
(function (GetBucketAclRequest) {
    GetBucketAclRequest.isa = function (o) {
        return __isa(o, "GetBucketAclRequest");
    };
})(GetBucketAclRequest || (GetBucketAclRequest = {}));
export var GetBucketAnalyticsConfigurationOutput;
(function (GetBucketAnalyticsConfigurationOutput) {
    GetBucketAnalyticsConfigurationOutput.isa = function (o) {
        return __isa(o, "GetBucketAnalyticsConfigurationOutput");
    };
})(GetBucketAnalyticsConfigurationOutput || (GetBucketAnalyticsConfigurationOutput = {}));
export var GetBucketAnalyticsConfigurationRequest;
(function (GetBucketAnalyticsConfigurationRequest) {
    GetBucketAnalyticsConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketAnalyticsConfigurationRequest");
    };
})(GetBucketAnalyticsConfigurationRequest || (GetBucketAnalyticsConfigurationRequest = {}));
export var GetBucketCorsOutput;
(function (GetBucketCorsOutput) {
    GetBucketCorsOutput.isa = function (o) {
        return __isa(o, "GetBucketCorsOutput");
    };
})(GetBucketCorsOutput || (GetBucketCorsOutput = {}));
export var GetBucketCorsRequest;
(function (GetBucketCorsRequest) {
    GetBucketCorsRequest.isa = function (o) {
        return __isa(o, "GetBucketCorsRequest");
    };
})(GetBucketCorsRequest || (GetBucketCorsRequest = {}));
export var GetBucketEncryptionOutput;
(function (GetBucketEncryptionOutput) {
    GetBucketEncryptionOutput.isa = function (o) {
        return __isa(o, "GetBucketEncryptionOutput");
    };
})(GetBucketEncryptionOutput || (GetBucketEncryptionOutput = {}));
export var GetBucketEncryptionRequest;
(function (GetBucketEncryptionRequest) {
    GetBucketEncryptionRequest.isa = function (o) {
        return __isa(o, "GetBucketEncryptionRequest");
    };
})(GetBucketEncryptionRequest || (GetBucketEncryptionRequest = {}));
export var GetBucketInventoryConfigurationOutput;
(function (GetBucketInventoryConfigurationOutput) {
    GetBucketInventoryConfigurationOutput.isa = function (o) {
        return __isa(o, "GetBucketInventoryConfigurationOutput");
    };
})(GetBucketInventoryConfigurationOutput || (GetBucketInventoryConfigurationOutput = {}));
export var GetBucketInventoryConfigurationRequest;
(function (GetBucketInventoryConfigurationRequest) {
    GetBucketInventoryConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketInventoryConfigurationRequest");
    };
})(GetBucketInventoryConfigurationRequest || (GetBucketInventoryConfigurationRequest = {}));
export var GetBucketLifecycleConfigurationOutput;
(function (GetBucketLifecycleConfigurationOutput) {
    GetBucketLifecycleConfigurationOutput.isa = function (o) {
        return __isa(o, "GetBucketLifecycleConfigurationOutput");
    };
})(GetBucketLifecycleConfigurationOutput || (GetBucketLifecycleConfigurationOutput = {}));
export var GetBucketLifecycleConfigurationRequest;
(function (GetBucketLifecycleConfigurationRequest) {
    GetBucketLifecycleConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketLifecycleConfigurationRequest");
    };
})(GetBucketLifecycleConfigurationRequest || (GetBucketLifecycleConfigurationRequest = {}));
export var GetBucketLocationOutput;
(function (GetBucketLocationOutput) {
    GetBucketLocationOutput.isa = function (o) {
        return __isa(o, "GetBucketLocationOutput");
    };
})(GetBucketLocationOutput || (GetBucketLocationOutput = {}));
export var GetBucketLocationRequest;
(function (GetBucketLocationRequest) {
    GetBucketLocationRequest.isa = function (o) {
        return __isa(o, "GetBucketLocationRequest");
    };
})(GetBucketLocationRequest || (GetBucketLocationRequest = {}));
export var GetBucketLoggingOutput;
(function (GetBucketLoggingOutput) {
    GetBucketLoggingOutput.isa = function (o) {
        return __isa(o, "GetBucketLoggingOutput");
    };
})(GetBucketLoggingOutput || (GetBucketLoggingOutput = {}));
export var GetBucketLoggingRequest;
(function (GetBucketLoggingRequest) {
    GetBucketLoggingRequest.isa = function (o) {
        return __isa(o, "GetBucketLoggingRequest");
    };
})(GetBucketLoggingRequest || (GetBucketLoggingRequest = {}));
export var GetBucketMetricsConfigurationOutput;
(function (GetBucketMetricsConfigurationOutput) {
    GetBucketMetricsConfigurationOutput.isa = function (o) {
        return __isa(o, "GetBucketMetricsConfigurationOutput");
    };
})(GetBucketMetricsConfigurationOutput || (GetBucketMetricsConfigurationOutput = {}));
export var GetBucketMetricsConfigurationRequest;
(function (GetBucketMetricsConfigurationRequest) {
    GetBucketMetricsConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketMetricsConfigurationRequest");
    };
})(GetBucketMetricsConfigurationRequest || (GetBucketMetricsConfigurationRequest = {}));
export var GetBucketNotificationConfigurationRequest;
(function (GetBucketNotificationConfigurationRequest) {
    GetBucketNotificationConfigurationRequest.isa = function (o) {
        return __isa(o, "GetBucketNotificationConfigurationRequest");
    };
})(GetBucketNotificationConfigurationRequest || (GetBucketNotificationConfigurationRequest = {}));
export var GetBucketPolicyOutput;
(function (GetBucketPolicyOutput) {
    GetBucketPolicyOutput.isa = function (o) {
        return __isa(o, "GetBucketPolicyOutput");
    };
})(GetBucketPolicyOutput || (GetBucketPolicyOutput = {}));
export var GetBucketPolicyRequest;
(function (GetBucketPolicyRequest) {
    GetBucketPolicyRequest.isa = function (o) {
        return __isa(o, "GetBucketPolicyRequest");
    };
})(GetBucketPolicyRequest || (GetBucketPolicyRequest = {}));
export var GetBucketPolicyStatusOutput;
(function (GetBucketPolicyStatusOutput) {
    GetBucketPolicyStatusOutput.isa = function (o) {
        return __isa(o, "GetBucketPolicyStatusOutput");
    };
})(GetBucketPolicyStatusOutput || (GetBucketPolicyStatusOutput = {}));
export var GetBucketPolicyStatusRequest;
(function (GetBucketPolicyStatusRequest) {
    GetBucketPolicyStatusRequest.isa = function (o) {
        return __isa(o, "GetBucketPolicyStatusRequest");
    };
})(GetBucketPolicyStatusRequest || (GetBucketPolicyStatusRequest = {}));
export var GetBucketReplicationOutput;
(function (GetBucketReplicationOutput) {
    GetBucketReplicationOutput.isa = function (o) {
        return __isa(o, "GetBucketReplicationOutput");
    };
})(GetBucketReplicationOutput || (GetBucketReplicationOutput = {}));
export var GetBucketReplicationRequest;
(function (GetBucketReplicationRequest) {
    GetBucketReplicationRequest.isa = function (o) {
        return __isa(o, "GetBucketReplicationRequest");
    };
})(GetBucketReplicationRequest || (GetBucketReplicationRequest = {}));
export var GetBucketRequestPaymentOutput;
(function (GetBucketRequestPaymentOutput) {
    GetBucketRequestPaymentOutput.isa = function (o) {
        return __isa(o, "GetBucketRequestPaymentOutput");
    };
})(GetBucketRequestPaymentOutput || (GetBucketRequestPaymentOutput = {}));
export var GetBucketRequestPaymentRequest;
(function (GetBucketRequestPaymentRequest) {
    GetBucketRequestPaymentRequest.isa = function (o) {
        return __isa(o, "GetBucketRequestPaymentRequest");
    };
})(GetBucketRequestPaymentRequest || (GetBucketRequestPaymentRequest = {}));
export var GetBucketTaggingOutput;
(function (GetBucketTaggingOutput) {
    GetBucketTaggingOutput.isa = function (o) {
        return __isa(o, "GetBucketTaggingOutput");
    };
})(GetBucketTaggingOutput || (GetBucketTaggingOutput = {}));
export var GetBucketTaggingRequest;
(function (GetBucketTaggingRequest) {
    GetBucketTaggingRequest.isa = function (o) {
        return __isa(o, "GetBucketTaggingRequest");
    };
})(GetBucketTaggingRequest || (GetBucketTaggingRequest = {}));
export var GetBucketVersioningOutput;
(function (GetBucketVersioningOutput) {
    GetBucketVersioningOutput.isa = function (o) {
        return __isa(o, "GetBucketVersioningOutput");
    };
})(GetBucketVersioningOutput || (GetBucketVersioningOutput = {}));
export var GetBucketVersioningRequest;
(function (GetBucketVersioningRequest) {
    GetBucketVersioningRequest.isa = function (o) {
        return __isa(o, "GetBucketVersioningRequest");
    };
})(GetBucketVersioningRequest || (GetBucketVersioningRequest = {}));
export var GetBucketWebsiteOutput;
(function (GetBucketWebsiteOutput) {
    GetBucketWebsiteOutput.isa = function (o) {
        return __isa(o, "GetBucketWebsiteOutput");
    };
})(GetBucketWebsiteOutput || (GetBucketWebsiteOutput = {}));
export var GetBucketWebsiteRequest;
(function (GetBucketWebsiteRequest) {
    GetBucketWebsiteRequest.isa = function (o) {
        return __isa(o, "GetBucketWebsiteRequest");
    };
})(GetBucketWebsiteRequest || (GetBucketWebsiteRequest = {}));
export var GetObjectAclOutput;
(function (GetObjectAclOutput) {
    GetObjectAclOutput.isa = function (o) {
        return __isa(o, "GetObjectAclOutput");
    };
})(GetObjectAclOutput || (GetObjectAclOutput = {}));
export var GetObjectAclRequest;
(function (GetObjectAclRequest) {
    GetObjectAclRequest.isa = function (o) {
        return __isa(o, "GetObjectAclRequest");
    };
})(GetObjectAclRequest || (GetObjectAclRequest = {}));
export var GetObjectLegalHoldOutput;
(function (GetObjectLegalHoldOutput) {
    GetObjectLegalHoldOutput.isa = function (o) {
        return __isa(o, "GetObjectLegalHoldOutput");
    };
})(GetObjectLegalHoldOutput || (GetObjectLegalHoldOutput = {}));
export var GetObjectLegalHoldRequest;
(function (GetObjectLegalHoldRequest) {
    GetObjectLegalHoldRequest.isa = function (o) {
        return __isa(o, "GetObjectLegalHoldRequest");
    };
})(GetObjectLegalHoldRequest || (GetObjectLegalHoldRequest = {}));
export var GetObjectLockConfigurationOutput;
(function (GetObjectLockConfigurationOutput) {
    GetObjectLockConfigurationOutput.isa = function (o) {
        return __isa(o, "GetObjectLockConfigurationOutput");
    };
})(GetObjectLockConfigurationOutput || (GetObjectLockConfigurationOutput = {}));
export var GetObjectLockConfigurationRequest;
(function (GetObjectLockConfigurationRequest) {
    GetObjectLockConfigurationRequest.isa = function (o) {
        return __isa(o, "GetObjectLockConfigurationRequest");
    };
})(GetObjectLockConfigurationRequest || (GetObjectLockConfigurationRequest = {}));
export var GetObjectOutput;
(function (GetObjectOutput) {
    GetObjectOutput.isa = function (o) {
        return __isa(o, "GetObjectOutput");
    };
})(GetObjectOutput || (GetObjectOutput = {}));
export var GetObjectRequest;
(function (GetObjectRequest) {
    GetObjectRequest.isa = function (o) {
        return __isa(o, "GetObjectRequest");
    };
})(GetObjectRequest || (GetObjectRequest = {}));
export var GetObjectRetentionOutput;
(function (GetObjectRetentionOutput) {
    GetObjectRetentionOutput.isa = function (o) {
        return __isa(o, "GetObjectRetentionOutput");
    };
})(GetObjectRetentionOutput || (GetObjectRetentionOutput = {}));
export var GetObjectRetentionRequest;
(function (GetObjectRetentionRequest) {
    GetObjectRetentionRequest.isa = function (o) {
        return __isa(o, "GetObjectRetentionRequest");
    };
})(GetObjectRetentionRequest || (GetObjectRetentionRequest = {}));
export var GetObjectTaggingOutput;
(function (GetObjectTaggingOutput) {
    GetObjectTaggingOutput.isa = function (o) {
        return __isa(o, "GetObjectTaggingOutput");
    };
})(GetObjectTaggingOutput || (GetObjectTaggingOutput = {}));
export var GetObjectTaggingRequest;
(function (GetObjectTaggingRequest) {
    GetObjectTaggingRequest.isa = function (o) {
        return __isa(o, "GetObjectTaggingRequest");
    };
})(GetObjectTaggingRequest || (GetObjectTaggingRequest = {}));
export var GetObjectTorrentOutput;
(function (GetObjectTorrentOutput) {
    GetObjectTorrentOutput.isa = function (o) {
        return __isa(o, "GetObjectTorrentOutput");
    };
})(GetObjectTorrentOutput || (GetObjectTorrentOutput = {}));
export var GetObjectTorrentRequest;
(function (GetObjectTorrentRequest) {
    GetObjectTorrentRequest.isa = function (o) {
        return __isa(o, "GetObjectTorrentRequest");
    };
})(GetObjectTorrentRequest || (GetObjectTorrentRequest = {}));
export var GetPublicAccessBlockOutput;
(function (GetPublicAccessBlockOutput) {
    GetPublicAccessBlockOutput.isa = function (o) {
        return __isa(o, "GetPublicAccessBlockOutput");
    };
})(GetPublicAccessBlockOutput || (GetPublicAccessBlockOutput = {}));
export var GetPublicAccessBlockRequest;
(function (GetPublicAccessBlockRequest) {
    GetPublicAccessBlockRequest.isa = function (o) {
        return __isa(o, "GetPublicAccessBlockRequest");
    };
})(GetPublicAccessBlockRequest || (GetPublicAccessBlockRequest = {}));
export var GlacierJobParameters;
(function (GlacierJobParameters) {
    GlacierJobParameters.isa = function (o) {
        return __isa(o, "GlacierJobParameters");
    };
})(GlacierJobParameters || (GlacierJobParameters = {}));
export var Grant;
(function (Grant) {
    Grant.isa = function (o) { return __isa(o, "Grant"); };
})(Grant || (Grant = {}));
export var Grantee;
(function (Grantee) {
    Grantee.isa = function (o) { return __isa(o, "Grantee"); };
})(Grantee || (Grantee = {}));
export var HeadBucketRequest;
(function (HeadBucketRequest) {
    HeadBucketRequest.isa = function (o) {
        return __isa(o, "HeadBucketRequest");
    };
})(HeadBucketRequest || (HeadBucketRequest = {}));
export var HeadObjectOutput;
(function (HeadObjectOutput) {
    HeadObjectOutput.isa = function (o) {
        return __isa(o, "HeadObjectOutput");
    };
})(HeadObjectOutput || (HeadObjectOutput = {}));
export var HeadObjectRequest;
(function (HeadObjectRequest) {
    HeadObjectRequest.isa = function (o) {
        return __isa(o, "HeadObjectRequest");
    };
})(HeadObjectRequest || (HeadObjectRequest = {}));
export var IndexDocument;
(function (IndexDocument) {
    IndexDocument.isa = function (o) { return __isa(o, "IndexDocument"); };
})(IndexDocument || (IndexDocument = {}));
export var Initiator;
(function (Initiator) {
    Initiator.isa = function (o) { return __isa(o, "Initiator"); };
})(Initiator || (Initiator = {}));
export var InputSerialization;
(function (InputSerialization) {
    InputSerialization.isa = function (o) {
        return __isa(o, "InputSerialization");
    };
})(InputSerialization || (InputSerialization = {}));
export var InventoryConfiguration;
(function (InventoryConfiguration) {
    InventoryConfiguration.isa = function (o) {
        return __isa(o, "InventoryConfiguration");
    };
})(InventoryConfiguration || (InventoryConfiguration = {}));
export var InventoryDestination;
(function (InventoryDestination) {
    InventoryDestination.isa = function (o) {
        return __isa(o, "InventoryDestination");
    };
})(InventoryDestination || (InventoryDestination = {}));
export var InventoryEncryption;
(function (InventoryEncryption) {
    InventoryEncryption.isa = function (o) {
        return __isa(o, "InventoryEncryption");
    };
})(InventoryEncryption || (InventoryEncryption = {}));
export var InventoryFilter;
(function (InventoryFilter) {
    InventoryFilter.isa = function (o) {
        return __isa(o, "InventoryFilter");
    };
})(InventoryFilter || (InventoryFilter = {}));
export var InventoryS3BucketDestination;
(function (InventoryS3BucketDestination) {
    InventoryS3BucketDestination.isa = function (o) {
        return __isa(o, "InventoryS3BucketDestination");
    };
})(InventoryS3BucketDestination || (InventoryS3BucketDestination = {}));
export var InventorySchedule;
(function (InventorySchedule) {
    InventorySchedule.isa = function (o) {
        return __isa(o, "InventorySchedule");
    };
})(InventorySchedule || (InventorySchedule = {}));
export var JSONInput;
(function (JSONInput) {
    JSONInput.isa = function (o) { return __isa(o, "JSONInput"); };
})(JSONInput || (JSONInput = {}));
export var JSONOutput;
(function (JSONOutput) {
    JSONOutput.isa = function (o) { return __isa(o, "JSONOutput"); };
})(JSONOutput || (JSONOutput = {}));
export var JSONType;
(function (JSONType) {
    JSONType["DOCUMENT"] = "DOCUMENT";
    JSONType["LINES"] = "LINES";
})(JSONType || (JSONType = {}));
export var LambdaFunctionConfiguration;
(function (LambdaFunctionConfiguration) {
    LambdaFunctionConfiguration.isa = function (o) {
        return __isa(o, "LambdaFunctionConfiguration");
    };
})(LambdaFunctionConfiguration || (LambdaFunctionConfiguration = {}));
export var LifecycleExpiration;
(function (LifecycleExpiration) {
    LifecycleExpiration.isa = function (o) {
        return __isa(o, "LifecycleExpiration");
    };
})(LifecycleExpiration || (LifecycleExpiration = {}));
export var LifecycleRule;
(function (LifecycleRule) {
    LifecycleRule.isa = function (o) { return __isa(o, "LifecycleRule"); };
})(LifecycleRule || (LifecycleRule = {}));
export var LifecycleRuleAndOperator;
(function (LifecycleRuleAndOperator) {
    LifecycleRuleAndOperator.isa = function (o) {
        return __isa(o, "LifecycleRuleAndOperator");
    };
})(LifecycleRuleAndOperator || (LifecycleRuleAndOperator = {}));
export var LifecycleRuleFilter;
(function (LifecycleRuleFilter) {
    LifecycleRuleFilter.isa = function (o) {
        return __isa(o, "LifecycleRuleFilter");
    };
})(LifecycleRuleFilter || (LifecycleRuleFilter = {}));
export var ListBucketAnalyticsConfigurationsOutput;
(function (ListBucketAnalyticsConfigurationsOutput) {
    ListBucketAnalyticsConfigurationsOutput.isa = function (o) {
        return __isa(o, "ListBucketAnalyticsConfigurationsOutput");
    };
})(ListBucketAnalyticsConfigurationsOutput || (ListBucketAnalyticsConfigurationsOutput = {}));
export var ListBucketAnalyticsConfigurationsRequest;
(function (ListBucketAnalyticsConfigurationsRequest) {
    ListBucketAnalyticsConfigurationsRequest.isa = function (o) {
        return __isa(o, "ListBucketAnalyticsConfigurationsRequest");
    };
})(ListBucketAnalyticsConfigurationsRequest || (ListBucketAnalyticsConfigurationsRequest = {}));
export var ListBucketInventoryConfigurationsOutput;
(function (ListBucketInventoryConfigurationsOutput) {
    ListBucketInventoryConfigurationsOutput.isa = function (o) {
        return __isa(o, "ListBucketInventoryConfigurationsOutput");
    };
})(ListBucketInventoryConfigurationsOutput || (ListBucketInventoryConfigurationsOutput = {}));
export var ListBucketInventoryConfigurationsRequest;
(function (ListBucketInventoryConfigurationsRequest) {
    ListBucketInventoryConfigurationsRequest.isa = function (o) {
        return __isa(o, "ListBucketInventoryConfigurationsRequest");
    };
})(ListBucketInventoryConfigurationsRequest || (ListBucketInventoryConfigurationsRequest = {}));
export var ListBucketMetricsConfigurationsOutput;
(function (ListBucketMetricsConfigurationsOutput) {
    ListBucketMetricsConfigurationsOutput.isa = function (o) {
        return __isa(o, "ListBucketMetricsConfigurationsOutput");
    };
})(ListBucketMetricsConfigurationsOutput || (ListBucketMetricsConfigurationsOutput = {}));
export var ListBucketMetricsConfigurationsRequest;
(function (ListBucketMetricsConfigurationsRequest) {
    ListBucketMetricsConfigurationsRequest.isa = function (o) {
        return __isa(o, "ListBucketMetricsConfigurationsRequest");
    };
})(ListBucketMetricsConfigurationsRequest || (ListBucketMetricsConfigurationsRequest = {}));
export var ListBucketsOutput;
(function (ListBucketsOutput) {
    ListBucketsOutput.isa = function (o) {
        return __isa(o, "ListBucketsOutput");
    };
})(ListBucketsOutput || (ListBucketsOutput = {}));
export var ListMultipartUploadsOutput;
(function (ListMultipartUploadsOutput) {
    ListMultipartUploadsOutput.isa = function (o) {
        return __isa(o, "ListMultipartUploadsOutput");
    };
})(ListMultipartUploadsOutput || (ListMultipartUploadsOutput = {}));
export var ListMultipartUploadsRequest;
(function (ListMultipartUploadsRequest) {
    ListMultipartUploadsRequest.isa = function (o) {
        return __isa(o, "ListMultipartUploadsRequest");
    };
})(ListMultipartUploadsRequest || (ListMultipartUploadsRequest = {}));
export var ListObjectVersionsOutput;
(function (ListObjectVersionsOutput) {
    ListObjectVersionsOutput.isa = function (o) {
        return __isa(o, "ListObjectVersionsOutput");
    };
})(ListObjectVersionsOutput || (ListObjectVersionsOutput = {}));
export var ListObjectVersionsRequest;
(function (ListObjectVersionsRequest) {
    ListObjectVersionsRequest.isa = function (o) {
        return __isa(o, "ListObjectVersionsRequest");
    };
})(ListObjectVersionsRequest || (ListObjectVersionsRequest = {}));
export var ListObjectsOutput;
(function (ListObjectsOutput) {
    ListObjectsOutput.isa = function (o) {
        return __isa(o, "ListObjectsOutput");
    };
})(ListObjectsOutput || (ListObjectsOutput = {}));
export var ListObjectsRequest;
(function (ListObjectsRequest) {
    ListObjectsRequest.isa = function (o) {
        return __isa(o, "ListObjectsRequest");
    };
})(ListObjectsRequest || (ListObjectsRequest = {}));
export var ListObjectsV2Output;
(function (ListObjectsV2Output) {
    ListObjectsV2Output.isa = function (o) {
        return __isa(o, "ListObjectsV2Output");
    };
})(ListObjectsV2Output || (ListObjectsV2Output = {}));
export var ListObjectsV2Request;
(function (ListObjectsV2Request) {
    ListObjectsV2Request.isa = function (o) {
        return __isa(o, "ListObjectsV2Request");
    };
})(ListObjectsV2Request || (ListObjectsV2Request = {}));
export var ListPartsOutput;
(function (ListPartsOutput) {
    ListPartsOutput.isa = function (o) {
        return __isa(o, "ListPartsOutput");
    };
})(ListPartsOutput || (ListPartsOutput = {}));
export var ListPartsRequest;
(function (ListPartsRequest) {
    ListPartsRequest.isa = function (o) {
        return __isa(o, "ListPartsRequest");
    };
})(ListPartsRequest || (ListPartsRequest = {}));
export var LoggingEnabled;
(function (LoggingEnabled) {
    LoggingEnabled.isa = function (o) {
        return __isa(o, "LoggingEnabled");
    };
})(LoggingEnabled || (LoggingEnabled = {}));
export var MetadataEntry;
(function (MetadataEntry) {
    MetadataEntry.isa = function (o) { return __isa(o, "MetadataEntry"); };
})(MetadataEntry || (MetadataEntry = {}));
export var Metrics;
(function (Metrics) {
    Metrics.isa = function (o) { return __isa(o, "Metrics"); };
})(Metrics || (Metrics = {}));
export var MetricsAndOperator;
(function (MetricsAndOperator) {
    MetricsAndOperator.isa = function (o) {
        return __isa(o, "MetricsAndOperator");
    };
})(MetricsAndOperator || (MetricsAndOperator = {}));
export var MetricsConfiguration;
(function (MetricsConfiguration) {
    MetricsConfiguration.isa = function (o) {
        return __isa(o, "MetricsConfiguration");
    };
})(MetricsConfiguration || (MetricsConfiguration = {}));
export var MetricsFilter;
(function (MetricsFilter) {
    MetricsFilter.isa = function (o) { return __isa(o, "MetricsFilter"); };
})(MetricsFilter || (MetricsFilter = {}));
export var MultipartUpload;
(function (MultipartUpload) {
    MultipartUpload.isa = function (o) {
        return __isa(o, "MultipartUpload");
    };
})(MultipartUpload || (MultipartUpload = {}));
export var NoSuchBucket;
(function (NoSuchBucket) {
    NoSuchBucket.isa = function (o) { return __isa(o, "NoSuchBucket"); };
})(NoSuchBucket || (NoSuchBucket = {}));
export var NoSuchKey;
(function (NoSuchKey) {
    NoSuchKey.isa = function (o) { return __isa(o, "NoSuchKey"); };
})(NoSuchKey || (NoSuchKey = {}));
export var NoSuchUpload;
(function (NoSuchUpload) {
    NoSuchUpload.isa = function (o) { return __isa(o, "NoSuchUpload"); };
})(NoSuchUpload || (NoSuchUpload = {}));
export var NoncurrentVersionExpiration;
(function (NoncurrentVersionExpiration) {
    NoncurrentVersionExpiration.isa = function (o) {
        return __isa(o, "NoncurrentVersionExpiration");
    };
})(NoncurrentVersionExpiration || (NoncurrentVersionExpiration = {}));
export var NoncurrentVersionTransition;
(function (NoncurrentVersionTransition) {
    NoncurrentVersionTransition.isa = function (o) {
        return __isa(o, "NoncurrentVersionTransition");
    };
})(NoncurrentVersionTransition || (NoncurrentVersionTransition = {}));
export var NotificationConfiguration;
(function (NotificationConfiguration) {
    NotificationConfiguration.isa = function (o) {
        return __isa(o, "NotificationConfiguration");
    };
})(NotificationConfiguration || (NotificationConfiguration = {}));
export var NotificationConfigurationFilter;
(function (NotificationConfigurationFilter) {
    NotificationConfigurationFilter.isa = function (o) {
        return __isa(o, "NotificationConfigurationFilter");
    };
})(NotificationConfigurationFilter || (NotificationConfigurationFilter = {}));
export var _Object;
(function (_Object) {
    _Object.isa = function (o) { return __isa(o, "Object"); };
})(_Object || (_Object = {}));
export var ObjectAlreadyInActiveTierError;
(function (ObjectAlreadyInActiveTierError) {
    ObjectAlreadyInActiveTierError.isa = function (o) {
        return __isa(o, "ObjectAlreadyInActiveTierError");
    };
})(ObjectAlreadyInActiveTierError || (ObjectAlreadyInActiveTierError = {}));
export var ObjectIdentifier;
(function (ObjectIdentifier) {
    ObjectIdentifier.isa = function (o) {
        return __isa(o, "ObjectIdentifier");
    };
})(ObjectIdentifier || (ObjectIdentifier = {}));
export var ObjectLockConfiguration;
(function (ObjectLockConfiguration) {
    ObjectLockConfiguration.isa = function (o) {
        return __isa(o, "ObjectLockConfiguration");
    };
})(ObjectLockConfiguration || (ObjectLockConfiguration = {}));
export var ObjectLockLegalHold;
(function (ObjectLockLegalHold) {
    ObjectLockLegalHold.isa = function (o) {
        return __isa(o, "ObjectLockLegalHold");
    };
})(ObjectLockLegalHold || (ObjectLockLegalHold = {}));
export var ObjectLockRetention;
(function (ObjectLockRetention) {
    ObjectLockRetention.isa = function (o) {
        return __isa(o, "ObjectLockRetention");
    };
})(ObjectLockRetention || (ObjectLockRetention = {}));
export var ObjectLockRule;
(function (ObjectLockRule) {
    ObjectLockRule.isa = function (o) {
        return __isa(o, "ObjectLockRule");
    };
})(ObjectLockRule || (ObjectLockRule = {}));
export var ObjectNotInActiveTierError;
(function (ObjectNotInActiveTierError) {
    ObjectNotInActiveTierError.isa = function (o) {
        return __isa(o, "ObjectNotInActiveTierError");
    };
})(ObjectNotInActiveTierError || (ObjectNotInActiveTierError = {}));
export var ObjectVersion;
(function (ObjectVersion) {
    ObjectVersion.isa = function (o) { return __isa(o, "ObjectVersion"); };
})(ObjectVersion || (ObjectVersion = {}));
export var OutputLocation;
(function (OutputLocation) {
    OutputLocation.isa = function (o) {
        return __isa(o, "OutputLocation");
    };
})(OutputLocation || (OutputLocation = {}));
export var OutputSerialization;
(function (OutputSerialization) {
    OutputSerialization.isa = function (o) {
        return __isa(o, "OutputSerialization");
    };
})(OutputSerialization || (OutputSerialization = {}));
export var Owner;
(function (Owner) {
    Owner.isa = function (o) { return __isa(o, "Owner"); };
})(Owner || (Owner = {}));
export var ParquetInput;
(function (ParquetInput) {
    ParquetInput.isa = function (o) { return __isa(o, "ParquetInput"); };
})(ParquetInput || (ParquetInput = {}));
export var Part;
(function (Part) {
    Part.isa = function (o) { return __isa(o, "Part"); };
})(Part || (Part = {}));
export var PolicyStatus;
(function (PolicyStatus) {
    PolicyStatus.isa = function (o) { return __isa(o, "PolicyStatus"); };
})(PolicyStatus || (PolicyStatus = {}));
export var Progress;
(function (Progress) {
    Progress.isa = function (o) { return __isa(o, "Progress"); };
})(Progress || (Progress = {}));
export var ProgressEvent;
(function (ProgressEvent) {
    ProgressEvent.isa = function (o) { return __isa(o, "ProgressEvent"); };
})(ProgressEvent || (ProgressEvent = {}));
export var PublicAccessBlockConfiguration;
(function (PublicAccessBlockConfiguration) {
    PublicAccessBlockConfiguration.isa = function (o) {
        return __isa(o, "PublicAccessBlockConfiguration");
    };
})(PublicAccessBlockConfiguration || (PublicAccessBlockConfiguration = {}));
export var PutBucketAccelerateConfigurationRequest;
(function (PutBucketAccelerateConfigurationRequest) {
    PutBucketAccelerateConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketAccelerateConfigurationRequest");
    };
})(PutBucketAccelerateConfigurationRequest || (PutBucketAccelerateConfigurationRequest = {}));
export var PutBucketAclRequest;
(function (PutBucketAclRequest) {
    PutBucketAclRequest.isa = function (o) {
        return __isa(o, "PutBucketAclRequest");
    };
})(PutBucketAclRequest || (PutBucketAclRequest = {}));
export var PutBucketAnalyticsConfigurationRequest;
(function (PutBucketAnalyticsConfigurationRequest) {
    PutBucketAnalyticsConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketAnalyticsConfigurationRequest");
    };
})(PutBucketAnalyticsConfigurationRequest || (PutBucketAnalyticsConfigurationRequest = {}));
export var PutBucketCorsRequest;
(function (PutBucketCorsRequest) {
    PutBucketCorsRequest.isa = function (o) {
        return __isa(o, "PutBucketCorsRequest");
    };
})(PutBucketCorsRequest || (PutBucketCorsRequest = {}));
export var PutBucketEncryptionRequest;
(function (PutBucketEncryptionRequest) {
    PutBucketEncryptionRequest.isa = function (o) {
        return __isa(o, "PutBucketEncryptionRequest");
    };
})(PutBucketEncryptionRequest || (PutBucketEncryptionRequest = {}));
export var PutBucketInventoryConfigurationRequest;
(function (PutBucketInventoryConfigurationRequest) {
    PutBucketInventoryConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketInventoryConfigurationRequest");
    };
})(PutBucketInventoryConfigurationRequest || (PutBucketInventoryConfigurationRequest = {}));
export var PutBucketLifecycleConfigurationRequest;
(function (PutBucketLifecycleConfigurationRequest) {
    PutBucketLifecycleConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketLifecycleConfigurationRequest");
    };
})(PutBucketLifecycleConfigurationRequest || (PutBucketLifecycleConfigurationRequest = {}));
export var PutBucketLoggingRequest;
(function (PutBucketLoggingRequest) {
    PutBucketLoggingRequest.isa = function (o) {
        return __isa(o, "PutBucketLoggingRequest");
    };
})(PutBucketLoggingRequest || (PutBucketLoggingRequest = {}));
export var PutBucketMetricsConfigurationRequest;
(function (PutBucketMetricsConfigurationRequest) {
    PutBucketMetricsConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketMetricsConfigurationRequest");
    };
})(PutBucketMetricsConfigurationRequest || (PutBucketMetricsConfigurationRequest = {}));
export var PutBucketNotificationConfigurationRequest;
(function (PutBucketNotificationConfigurationRequest) {
    PutBucketNotificationConfigurationRequest.isa = function (o) {
        return __isa(o, "PutBucketNotificationConfigurationRequest");
    };
})(PutBucketNotificationConfigurationRequest || (PutBucketNotificationConfigurationRequest = {}));
export var PutBucketPolicyRequest;
(function (PutBucketPolicyRequest) {
    PutBucketPolicyRequest.isa = function (o) {
        return __isa(o, "PutBucketPolicyRequest");
    };
})(PutBucketPolicyRequest || (PutBucketPolicyRequest = {}));
export var PutBucketReplicationRequest;
(function (PutBucketReplicationRequest) {
    PutBucketReplicationRequest.isa = function (o) {
        return __isa(o, "PutBucketReplicationRequest");
    };
})(PutBucketReplicationRequest || (PutBucketReplicationRequest = {}));
export var PutBucketRequestPaymentRequest;
(function (PutBucketRequestPaymentRequest) {
    PutBucketRequestPaymentRequest.isa = function (o) {
        return __isa(o, "PutBucketRequestPaymentRequest");
    };
})(PutBucketRequestPaymentRequest || (PutBucketRequestPaymentRequest = {}));
export var PutBucketTaggingRequest;
(function (PutBucketTaggingRequest) {
    PutBucketTaggingRequest.isa = function (o) {
        return __isa(o, "PutBucketTaggingRequest");
    };
})(PutBucketTaggingRequest || (PutBucketTaggingRequest = {}));
export var PutBucketVersioningRequest;
(function (PutBucketVersioningRequest) {
    PutBucketVersioningRequest.isa = function (o) {
        return __isa(o, "PutBucketVersioningRequest");
    };
})(PutBucketVersioningRequest || (PutBucketVersioningRequest = {}));
export var PutBucketWebsiteRequest;
(function (PutBucketWebsiteRequest) {
    PutBucketWebsiteRequest.isa = function (o) {
        return __isa(o, "PutBucketWebsiteRequest");
    };
})(PutBucketWebsiteRequest || (PutBucketWebsiteRequest = {}));
export var PutObjectAclOutput;
(function (PutObjectAclOutput) {
    PutObjectAclOutput.isa = function (o) {
        return __isa(o, "PutObjectAclOutput");
    };
})(PutObjectAclOutput || (PutObjectAclOutput = {}));
export var PutObjectAclRequest;
(function (PutObjectAclRequest) {
    PutObjectAclRequest.isa = function (o) {
        return __isa(o, "PutObjectAclRequest");
    };
})(PutObjectAclRequest || (PutObjectAclRequest = {}));
export var PutObjectLegalHoldOutput;
(function (PutObjectLegalHoldOutput) {
    PutObjectLegalHoldOutput.isa = function (o) {
        return __isa(o, "PutObjectLegalHoldOutput");
    };
})(PutObjectLegalHoldOutput || (PutObjectLegalHoldOutput = {}));
export var PutObjectLegalHoldRequest;
(function (PutObjectLegalHoldRequest) {
    PutObjectLegalHoldRequest.isa = function (o) {
        return __isa(o, "PutObjectLegalHoldRequest");
    };
})(PutObjectLegalHoldRequest || (PutObjectLegalHoldRequest = {}));
export var PutObjectLockConfigurationOutput;
(function (PutObjectLockConfigurationOutput) {
    PutObjectLockConfigurationOutput.isa = function (o) {
        return __isa(o, "PutObjectLockConfigurationOutput");
    };
})(PutObjectLockConfigurationOutput || (PutObjectLockConfigurationOutput = {}));
export var PutObjectLockConfigurationRequest;
(function (PutObjectLockConfigurationRequest) {
    PutObjectLockConfigurationRequest.isa = function (o) {
        return __isa(o, "PutObjectLockConfigurationRequest");
    };
})(PutObjectLockConfigurationRequest || (PutObjectLockConfigurationRequest = {}));
export var PutObjectOutput;
(function (PutObjectOutput) {
    PutObjectOutput.isa = function (o) {
        return __isa(o, "PutObjectOutput");
    };
})(PutObjectOutput || (PutObjectOutput = {}));
export var PutObjectRequest;
(function (PutObjectRequest) {
    PutObjectRequest.isa = function (o) {
        return __isa(o, "PutObjectRequest");
    };
})(PutObjectRequest || (PutObjectRequest = {}));
export var PutObjectRetentionOutput;
(function (PutObjectRetentionOutput) {
    PutObjectRetentionOutput.isa = function (o) {
        return __isa(o, "PutObjectRetentionOutput");
    };
})(PutObjectRetentionOutput || (PutObjectRetentionOutput = {}));
export var PutObjectRetentionRequest;
(function (PutObjectRetentionRequest) {
    PutObjectRetentionRequest.isa = function (o) {
        return __isa(o, "PutObjectRetentionRequest");
    };
})(PutObjectRetentionRequest || (PutObjectRetentionRequest = {}));
export var PutObjectTaggingOutput;
(function (PutObjectTaggingOutput) {
    PutObjectTaggingOutput.isa = function (o) {
        return __isa(o, "PutObjectTaggingOutput");
    };
})(PutObjectTaggingOutput || (PutObjectTaggingOutput = {}));
export var PutObjectTaggingRequest;
(function (PutObjectTaggingRequest) {
    PutObjectTaggingRequest.isa = function (o) {
        return __isa(o, "PutObjectTaggingRequest");
    };
})(PutObjectTaggingRequest || (PutObjectTaggingRequest = {}));
export var PutPublicAccessBlockRequest;
(function (PutPublicAccessBlockRequest) {
    PutPublicAccessBlockRequest.isa = function (o) {
        return __isa(o, "PutPublicAccessBlockRequest");
    };
})(PutPublicAccessBlockRequest || (PutPublicAccessBlockRequest = {}));
export var QueueConfiguration;
(function (QueueConfiguration) {
    QueueConfiguration.isa = function (o) {
        return __isa(o, "QueueConfiguration");
    };
})(QueueConfiguration || (QueueConfiguration = {}));
export var QuoteFields;
(function (QuoteFields) {
    QuoteFields["ALWAYS"] = "ALWAYS";
    QuoteFields["ASNEEDED"] = "ASNEEDED";
})(QuoteFields || (QuoteFields = {}));
export var RecordsEvent;
(function (RecordsEvent) {
    RecordsEvent.isa = function (o) { return __isa(o, "RecordsEvent"); };
})(RecordsEvent || (RecordsEvent = {}));
export var Redirect;
(function (Redirect) {
    Redirect.isa = function (o) { return __isa(o, "Redirect"); };
})(Redirect || (Redirect = {}));
export var RedirectAllRequestsTo;
(function (RedirectAllRequestsTo) {
    RedirectAllRequestsTo.isa = function (o) {
        return __isa(o, "RedirectAllRequestsTo");
    };
})(RedirectAllRequestsTo || (RedirectAllRequestsTo = {}));
export var ReplicationConfiguration;
(function (ReplicationConfiguration) {
    ReplicationConfiguration.isa = function (o) {
        return __isa(o, "ReplicationConfiguration");
    };
})(ReplicationConfiguration || (ReplicationConfiguration = {}));
export var ReplicationRule;
(function (ReplicationRule) {
    ReplicationRule.isa = function (o) {
        return __isa(o, "ReplicationRule");
    };
})(ReplicationRule || (ReplicationRule = {}));
export var ReplicationRuleAndOperator;
(function (ReplicationRuleAndOperator) {
    ReplicationRuleAndOperator.isa = function (o) {
        return __isa(o, "ReplicationRuleAndOperator");
    };
})(ReplicationRuleAndOperator || (ReplicationRuleAndOperator = {}));
export var ReplicationRuleFilter;
(function (ReplicationRuleFilter) {
    ReplicationRuleFilter.isa = function (o) {
        return __isa(o, "ReplicationRuleFilter");
    };
})(ReplicationRuleFilter || (ReplicationRuleFilter = {}));
export var ReplicationTime;
(function (ReplicationTime) {
    ReplicationTime.isa = function (o) {
        return __isa(o, "ReplicationTime");
    };
})(ReplicationTime || (ReplicationTime = {}));
export var ReplicationTimeValue;
(function (ReplicationTimeValue) {
    ReplicationTimeValue.isa = function (o) {
        return __isa(o, "ReplicationTimeValue");
    };
})(ReplicationTimeValue || (ReplicationTimeValue = {}));
export var RequestPaymentConfiguration;
(function (RequestPaymentConfiguration) {
    RequestPaymentConfiguration.isa = function (o) {
        return __isa(o, "RequestPaymentConfiguration");
    };
})(RequestPaymentConfiguration || (RequestPaymentConfiguration = {}));
export var RequestProgress;
(function (RequestProgress) {
    RequestProgress.isa = function (o) {
        return __isa(o, "RequestProgress");
    };
})(RequestProgress || (RequestProgress = {}));
export var RestoreObjectOutput;
(function (RestoreObjectOutput) {
    RestoreObjectOutput.isa = function (o) {
        return __isa(o, "RestoreObjectOutput");
    };
})(RestoreObjectOutput || (RestoreObjectOutput = {}));
export var RestoreObjectRequest;
(function (RestoreObjectRequest) {
    RestoreObjectRequest.isa = function (o) {
        return __isa(o, "RestoreObjectRequest");
    };
})(RestoreObjectRequest || (RestoreObjectRequest = {}));
export var RestoreRequest;
(function (RestoreRequest) {
    RestoreRequest.isa = function (o) {
        return __isa(o, "RestoreRequest");
    };
})(RestoreRequest || (RestoreRequest = {}));
export var RestoreRequestType;
(function (RestoreRequestType) {
    RestoreRequestType["SELECT"] = "SELECT";
})(RestoreRequestType || (RestoreRequestType = {}));
export var RoutingRule;
(function (RoutingRule) {
    RoutingRule.isa = function (o) { return __isa(o, "RoutingRule"); };
})(RoutingRule || (RoutingRule = {}));
export var S3KeyFilter;
(function (S3KeyFilter) {
    S3KeyFilter.isa = function (o) { return __isa(o, "S3KeyFilter"); };
})(S3KeyFilter || (S3KeyFilter = {}));
export var S3Location;
(function (S3Location) {
    S3Location.isa = function (o) { return __isa(o, "S3Location"); };
})(S3Location || (S3Location = {}));
export var SSEKMS;
(function (SSEKMS) {
    SSEKMS.isa = function (o) { return __isa(o, "SSEKMS"); };
})(SSEKMS || (SSEKMS = {}));
export var SSES3;
(function (SSES3) {
    SSES3.isa = function (o) { return __isa(o, "SSES3"); };
})(SSES3 || (SSES3 = {}));
export var ScanRange;
(function (ScanRange) {
    ScanRange.isa = function (o) { return __isa(o, "ScanRange"); };
})(ScanRange || (ScanRange = {}));
export var SelectObjectContentEventStream;
(function (SelectObjectContentEventStream) {
    SelectObjectContentEventStream.visit = function (value, visitor) {
        if (value.Cont !== undefined)
            return visitor.Cont(value.Cont);
        if (value.End !== undefined)
            return visitor.End(value.End);
        if (value.Progress !== undefined)
            return visitor.Progress(value.Progress);
        if (value.Records !== undefined)
            return visitor.Records(value.Records);
        if (value.Stats !== undefined)
            return visitor.Stats(value.Stats);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
})(SelectObjectContentEventStream || (SelectObjectContentEventStream = {}));
export var SelectObjectContentOutput;
(function (SelectObjectContentOutput) {
    SelectObjectContentOutput.isa = function (o) {
        return __isa(o, "SelectObjectContentOutput");
    };
})(SelectObjectContentOutput || (SelectObjectContentOutput = {}));
export var SelectObjectContentRequest;
(function (SelectObjectContentRequest) {
    SelectObjectContentRequest.isa = function (o) {
        return __isa(o, "SelectObjectContentRequest");
    };
})(SelectObjectContentRequest || (SelectObjectContentRequest = {}));
export var SelectParameters;
(function (SelectParameters) {
    SelectParameters.isa = function (o) {
        return __isa(o, "SelectParameters");
    };
})(SelectParameters || (SelectParameters = {}));
export var ServerSideEncryptionByDefault;
(function (ServerSideEncryptionByDefault) {
    ServerSideEncryptionByDefault.isa = function (o) {
        return __isa(o, "ServerSideEncryptionByDefault");
    };
})(ServerSideEncryptionByDefault || (ServerSideEncryptionByDefault = {}));
export var ServerSideEncryptionConfiguration;
(function (ServerSideEncryptionConfiguration) {
    ServerSideEncryptionConfiguration.isa = function (o) {
        return __isa(o, "ServerSideEncryptionConfiguration");
    };
})(ServerSideEncryptionConfiguration || (ServerSideEncryptionConfiguration = {}));
export var ServerSideEncryptionRule;
(function (ServerSideEncryptionRule) {
    ServerSideEncryptionRule.isa = function (o) {
        return __isa(o, "ServerSideEncryptionRule");
    };
})(ServerSideEncryptionRule || (ServerSideEncryptionRule = {}));
export var SourceSelectionCriteria;
(function (SourceSelectionCriteria) {
    SourceSelectionCriteria.isa = function (o) {
        return __isa(o, "SourceSelectionCriteria");
    };
})(SourceSelectionCriteria || (SourceSelectionCriteria = {}));
export var SseKmsEncryptedObjects;
(function (SseKmsEncryptedObjects) {
    SseKmsEncryptedObjects.isa = function (o) {
        return __isa(o, "SseKmsEncryptedObjects");
    };
})(SseKmsEncryptedObjects || (SseKmsEncryptedObjects = {}));
export var Stats;
(function (Stats) {
    Stats.isa = function (o) { return __isa(o, "Stats"); };
})(Stats || (Stats = {}));
export var StatsEvent;
(function (StatsEvent) {
    StatsEvent.isa = function (o) { return __isa(o, "StatsEvent"); };
})(StatsEvent || (StatsEvent = {}));
export var StorageClassAnalysis;
(function (StorageClassAnalysis) {
    StorageClassAnalysis.isa = function (o) {
        return __isa(o, "StorageClassAnalysis");
    };
})(StorageClassAnalysis || (StorageClassAnalysis = {}));
export var StorageClassAnalysisDataExport;
(function (StorageClassAnalysisDataExport) {
    StorageClassAnalysisDataExport.isa = function (o) {
        return __isa(o, "StorageClassAnalysisDataExport");
    };
})(StorageClassAnalysisDataExport || (StorageClassAnalysisDataExport = {}));
export var Tag;
(function (Tag) {
    Tag.isa = function (o) { return __isa(o, "Tag"); };
})(Tag || (Tag = {}));
export var Tagging;
(function (Tagging) {
    Tagging.isa = function (o) { return __isa(o, "Tagging"); };
})(Tagging || (Tagging = {}));
export var TargetGrant;
(function (TargetGrant) {
    TargetGrant.isa = function (o) { return __isa(o, "TargetGrant"); };
})(TargetGrant || (TargetGrant = {}));
export var TopicConfiguration;
(function (TopicConfiguration) {
    TopicConfiguration.isa = function (o) {
        return __isa(o, "TopicConfiguration");
    };
})(TopicConfiguration || (TopicConfiguration = {}));
export var Transition;
(function (Transition) {
    Transition.isa = function (o) { return __isa(o, "Transition"); };
})(Transition || (Transition = {}));
export var UploadPartCopyOutput;
(function (UploadPartCopyOutput) {
    UploadPartCopyOutput.isa = function (o) {
        return __isa(o, "UploadPartCopyOutput");
    };
})(UploadPartCopyOutput || (UploadPartCopyOutput = {}));
export var UploadPartCopyRequest;
(function (UploadPartCopyRequest) {
    UploadPartCopyRequest.isa = function (o) {
        return __isa(o, "UploadPartCopyRequest");
    };
})(UploadPartCopyRequest || (UploadPartCopyRequest = {}));
export var UploadPartOutput;
(function (UploadPartOutput) {
    UploadPartOutput.isa = function (o) {
        return __isa(o, "UploadPartOutput");
    };
})(UploadPartOutput || (UploadPartOutput = {}));
export var UploadPartRequest;
(function (UploadPartRequest) {
    UploadPartRequest.isa = function (o) {
        return __isa(o, "UploadPartRequest");
    };
})(UploadPartRequest || (UploadPartRequest = {}));
export var VersioningConfiguration;
(function (VersioningConfiguration) {
    VersioningConfiguration.isa = function (o) {
        return __isa(o, "VersioningConfiguration");
    };
})(VersioningConfiguration || (VersioningConfiguration = {}));
export var WebsiteConfiguration;
(function (WebsiteConfiguration) {
    WebsiteConfiguration.isa = function (o) {
        return __isa(o, "WebsiteConfiguration");
    };
})(WebsiteConfiguration || (WebsiteConfiguration = {}));
