// @aws-sdk/client-textract/dist/es/models/index.js
import { isa as __isa } from "@aws-sdk/smithy-client";
export var AccessDeniedException;
(function (AccessDeniedException) {
    AccessDeniedException.isa = function (o) {
        return __isa(o, "AccessDeniedException");
    };
})(AccessDeniedException || (AccessDeniedException = {}));
export var AnalyzeDocumentRequest;
(function (AnalyzeDocumentRequest) {
    AnalyzeDocumentRequest.isa = function (o) {
        return __isa(o, "AnalyzeDocumentRequest");
    };
})(AnalyzeDocumentRequest || (AnalyzeDocumentRequest = {}));
export var AnalyzeDocumentResponse;
(function (AnalyzeDocumentResponse) {
    AnalyzeDocumentResponse.isa = function (o) {
        return __isa(o, "AnalyzeDocumentResponse");
    };
})(AnalyzeDocumentResponse || (AnalyzeDocumentResponse = {}));
export var BadDocumentException;
(function (BadDocumentException) {
    BadDocumentException.isa = function (o) {
        return __isa(o, "BadDocumentException");
    };
})(BadDocumentException || (BadDocumentException = {}));
export var Block;
(function (Block) {
    Block.isa = function (o) { return __isa(o, "Block"); };
})(Block || (Block = {}));
export var BlockType;
(function (BlockType) {
    BlockType["CELL"] = "CELL";
    BlockType["KEY_VALUE_SET"] = "KEY_VALUE_SET";
    BlockType["LINE"] = "LINE";
    BlockType["PAGE"] = "PAGE";
    BlockType["SELECTION_ELEMENT"] = "SELECTION_ELEMENT";
    BlockType["TABLE"] = "TABLE";
    BlockType["WORD"] = "WORD";
})(BlockType || (BlockType = {}));
export var BoundingBox;
(function (BoundingBox) {
    BoundingBox.isa = function (o) { return __isa(o, "BoundingBox"); };
})(BoundingBox || (BoundingBox = {}));
export var ContentClassifier;
(function (ContentClassifier) {
    ContentClassifier["FREE_OF_ADULT_CONTENT"] = "FreeOfAdultContent";
    ContentClassifier["FREE_OF_PERSONALLY_IDENTIFIABLE_INFORMATION"] = "FreeOfPersonallyIdentifiableInformation";
})(ContentClassifier || (ContentClassifier = {}));
export var DetectDocumentTextRequest;
(function (DetectDocumentTextRequest) {
    DetectDocumentTextRequest.isa = function (o) {
        return __isa(o, "DetectDocumentTextRequest");
    };
})(DetectDocumentTextRequest || (DetectDocumentTextRequest = {}));
export var DetectDocumentTextResponse;
(function (DetectDocumentTextResponse) {
    DetectDocumentTextResponse.isa = function (o) {
        return __isa(o, "DetectDocumentTextResponse");
    };
})(DetectDocumentTextResponse || (DetectDocumentTextResponse = {}));
export var Document;
(function (Document) {
    Document.isa = function (o) { return __isa(o, "Document"); };
})(Document || (Document = {}));
export var DocumentLocation;
(function (DocumentLocation) {
    DocumentLocation.isa = function (o) {
        return __isa(o, "DocumentLocation");
    };
})(DocumentLocation || (DocumentLocation = {}));
export var DocumentMetadata;
(function (DocumentMetadata) {
    DocumentMetadata.isa = function (o) {
        return __isa(o, "DocumentMetadata");
    };
})(DocumentMetadata || (DocumentMetadata = {}));
export var DocumentTooLargeException;
(function (DocumentTooLargeException) {
    DocumentTooLargeException.isa = function (o) {
        return __isa(o, "DocumentTooLargeException");
    };
})(DocumentTooLargeException || (DocumentTooLargeException = {}));
export var EntityType;
(function (EntityType) {
    EntityType["KEY"] = "KEY";
    EntityType["VALUE"] = "VALUE";
})(EntityType || (EntityType = {}));
export var FeatureType;
(function (FeatureType) {
    FeatureType["FORMS"] = "FORMS";
    FeatureType["TABLES"] = "TABLES";
})(FeatureType || (FeatureType = {}));
export var Geometry;
(function (Geometry) {
    Geometry.isa = function (o) { return __isa(o, "Geometry"); };
})(Geometry || (Geometry = {}));
export var GetDocumentAnalysisRequest;
(function (GetDocumentAnalysisRequest) {
    GetDocumentAnalysisRequest.isa = function (o) {
        return __isa(o, "GetDocumentAnalysisRequest");
    };
})(GetDocumentAnalysisRequest || (GetDocumentAnalysisRequest = {}));
export var GetDocumentAnalysisResponse;
(function (GetDocumentAnalysisResponse) {
    GetDocumentAnalysisResponse.isa = function (o) {
        return __isa(o, "GetDocumentAnalysisResponse");
    };
})(GetDocumentAnalysisResponse || (GetDocumentAnalysisResponse = {}));
export var GetDocumentTextDetectionRequest;
(function (GetDocumentTextDetectionRequest) {
    GetDocumentTextDetectionRequest.isa = function (o) {
        return __isa(o, "GetDocumentTextDetectionRequest");
    };
})(GetDocumentTextDetectionRequest || (GetDocumentTextDetectionRequest = {}));
export var GetDocumentTextDetectionResponse;
(function (GetDocumentTextDetectionResponse) {
    GetDocumentTextDetectionResponse.isa = function (o) {
        return __isa(o, "GetDocumentTextDetectionResponse");
    };
})(GetDocumentTextDetectionResponse || (GetDocumentTextDetectionResponse = {}));
export var HumanLoopActivationOutput;
(function (HumanLoopActivationOutput) {
    HumanLoopActivationOutput.isa = function (o) {
        return __isa(o, "HumanLoopActivationOutput");
    };
})(HumanLoopActivationOutput || (HumanLoopActivationOutput = {}));
export var HumanLoopConfig;
(function (HumanLoopConfig) {
    HumanLoopConfig.isa = function (o) {
        return __isa(o, "HumanLoopConfig");
    };
})(HumanLoopConfig || (HumanLoopConfig = {}));
export var HumanLoopDataAttributes;
(function (HumanLoopDataAttributes) {
    HumanLoopDataAttributes.isa = function (o) {
        return __isa(o, "HumanLoopDataAttributes");
    };
})(HumanLoopDataAttributes || (HumanLoopDataAttributes = {}));
export var HumanLoopQuotaExceededException;
(function (HumanLoopQuotaExceededException) {
    HumanLoopQuotaExceededException.isa = function (o) {
        return __isa(o, "HumanLoopQuotaExceededException");
    };
})(HumanLoopQuotaExceededException || (HumanLoopQuotaExceededException = {}));
export var IdempotentParameterMismatchException;
(function (IdempotentParameterMismatchException) {
    IdempotentParameterMismatchException.isa = function (o) {
        return __isa(o, "IdempotentParameterMismatchException");
    };
})(IdempotentParameterMismatchException || (IdempotentParameterMismatchException = {}));
export var InternalServerError;
(function (InternalServerError) {
    InternalServerError.isa = function (o) {
        return __isa(o, "InternalServerError");
    };
})(InternalServerError || (InternalServerError = {}));
export var InvalidJobIdException;
(function (InvalidJobIdException) {
    InvalidJobIdException.isa = function (o) {
        return __isa(o, "InvalidJobIdException");
    };
})(InvalidJobIdException || (InvalidJobIdException = {}));
export var InvalidParameterException;
(function (InvalidParameterException) {
    InvalidParameterException.isa = function (o) {
        return __isa(o, "InvalidParameterException");
    };
})(InvalidParameterException || (InvalidParameterException = {}));
export var InvalidS3ObjectException;
(function (InvalidS3ObjectException) {
    InvalidS3ObjectException.isa = function (o) {
        return __isa(o, "InvalidS3ObjectException");
    };
})(InvalidS3ObjectException || (InvalidS3ObjectException = {}));
export var JobStatus;
(function (JobStatus) {
    JobStatus["FAILED"] = "FAILED";
    JobStatus["IN_PROGRESS"] = "IN_PROGRESS";
    JobStatus["PARTIAL_SUCCESS"] = "PARTIAL_SUCCESS";
    JobStatus["SUCCEEDED"] = "SUCCEEDED";
})(JobStatus || (JobStatus = {}));
export var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.isa = function (o) {
        return __isa(o, "LimitExceededException");
    };
})(LimitExceededException || (LimitExceededException = {}));
export var NotificationChannel;
(function (NotificationChannel) {
    NotificationChannel.isa = function (o) {
        return __isa(o, "NotificationChannel");
    };
})(NotificationChannel || (NotificationChannel = {}));
export var Point;
(function (Point) {
    Point.isa = function (o) { return __isa(o, "Point"); };
})(Point || (Point = {}));
export var ProvisionedThroughputExceededException;
(function (ProvisionedThroughputExceededException) {
    ProvisionedThroughputExceededException.isa = function (o) {
        return __isa(o, "ProvisionedThroughputExceededException");
    };
})(ProvisionedThroughputExceededException || (ProvisionedThroughputExceededException = {}));
export var Relationship;
(function (Relationship) {
    Relationship.isa = function (o) { return __isa(o, "Relationship"); };
})(Relationship || (Relationship = {}));
export var RelationshipType;
(function (RelationshipType) {
    RelationshipType["CHILD"] = "CHILD";
    RelationshipType["VALUE"] = "VALUE";
})(RelationshipType || (RelationshipType = {}));
export var S3Object;
(function (S3Object) {
    S3Object.isa = function (o) { return __isa(o, "S3Object"); };
})(S3Object || (S3Object = {}));
export var SelectionStatus;
(function (SelectionStatus) {
    SelectionStatus["NOT_SELECTED"] = "NOT_SELECTED";
    SelectionStatus["SELECTED"] = "SELECTED";
})(SelectionStatus || (SelectionStatus = {}));
export var StartDocumentAnalysisRequest;
(function (StartDocumentAnalysisRequest) {
    StartDocumentAnalysisRequest.isa = function (o) {
        return __isa(o, "StartDocumentAnalysisRequest");
    };
})(StartDocumentAnalysisRequest || (StartDocumentAnalysisRequest = {}));
export var StartDocumentAnalysisResponse;
(function (StartDocumentAnalysisResponse) {
    StartDocumentAnalysisResponse.isa = function (o) {
        return __isa(o, "StartDocumentAnalysisResponse");
    };
})(StartDocumentAnalysisResponse || (StartDocumentAnalysisResponse = {}));
export var StartDocumentTextDetectionRequest;
(function (StartDocumentTextDetectionRequest) {
    StartDocumentTextDetectionRequest.isa = function (o) {
        return __isa(o, "StartDocumentTextDetectionRequest");
    };
})(StartDocumentTextDetectionRequest || (StartDocumentTextDetectionRequest = {}));
export var StartDocumentTextDetectionResponse;
(function (StartDocumentTextDetectionResponse) {
    StartDocumentTextDetectionResponse.isa = function (o) {
        return __isa(o, "StartDocumentTextDetectionResponse");
    };
})(StartDocumentTextDetectionResponse || (StartDocumentTextDetectionResponse = {}));
export var ThrottlingException;
(function (ThrottlingException) {
    ThrottlingException.isa = function (o) {
        return __isa(o, "ThrottlingException");
    };
})(ThrottlingException || (ThrottlingException = {}));
export var UnsupportedDocumentException;
(function (UnsupportedDocumentException) {
    UnsupportedDocumentException.isa = function (o) {
        return __isa(o, "UnsupportedDocumentException");
    };
})(UnsupportedDocumentException || (UnsupportedDocumentException = {}));
export var Warning;
(function (Warning) {
    Warning.isa = function (o) { return __isa(o, "Warning"); };
})(Warning || (Warning = {}));
