// @aws-sdk/client-polly/dist/cjs/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const smithy_client_1 = require("@aws-sdk/smithy-client");
var DeleteLexiconInput;
(function (DeleteLexiconInput) {
    DeleteLexiconInput.isa = (o) => smithy_client_1.isa(o, "DeleteLexiconInput");
})(DeleteLexiconInput = exports.DeleteLexiconInput || (exports.DeleteLexiconInput = {}));
var DeleteLexiconOutput;
(function (DeleteLexiconOutput) {
    DeleteLexiconOutput.isa = (o) => smithy_client_1.isa(o, "DeleteLexiconOutput");
})(DeleteLexiconOutput = exports.DeleteLexiconOutput || (exports.DeleteLexiconOutput = {}));
var DescribeVoicesInput;
(function (DescribeVoicesInput) {
    DescribeVoicesInput.isa = (o) => smithy_client_1.isa(o, "DescribeVoicesInput");
})(DescribeVoicesInput = exports.DescribeVoicesInput || (exports.DescribeVoicesInput = {}));
var DescribeVoicesOutput;
(function (DescribeVoicesOutput) {
    DescribeVoicesOutput.isa = (o) => smithy_client_1.isa(o, "DescribeVoicesOutput");
})(DescribeVoicesOutput = exports.DescribeVoicesOutput || (exports.DescribeVoicesOutput = {}));
var Engine;
(function (Engine) {
    Engine["NEURAL"] = "neural";
    Engine["STANDARD"] = "standard";
})(Engine = exports.Engine || (exports.Engine = {}));
var EngineNotSupportedException;
(function (EngineNotSupportedException) {
    EngineNotSupportedException.isa = (o) => smithy_client_1.isa(o, "EngineNotSupportedException");
})(EngineNotSupportedException = exports.EngineNotSupportedException || (exports.EngineNotSupportedException = {}));
var GetLexiconInput;
(function (GetLexiconInput) {
    GetLexiconInput.isa = (o) => smithy_client_1.isa(o, "GetLexiconInput");
})(GetLexiconInput = exports.GetLexiconInput || (exports.GetLexiconInput = {}));
var GetLexiconOutput;
(function (GetLexiconOutput) {
    GetLexiconOutput.isa = (o) => smithy_client_1.isa(o, "GetLexiconOutput");
})(GetLexiconOutput = exports.GetLexiconOutput || (exports.GetLexiconOutput = {}));
var GetSpeechSynthesisTaskInput;
(function (GetSpeechSynthesisTaskInput) {
    GetSpeechSynthesisTaskInput.isa = (o) => smithy_client_1.isa(o, "GetSpeechSynthesisTaskInput");
})(GetSpeechSynthesisTaskInput = exports.GetSpeechSynthesisTaskInput || (exports.GetSpeechSynthesisTaskInput = {}));
var GetSpeechSynthesisTaskOutput;
(function (GetSpeechSynthesisTaskOutput) {
    GetSpeechSynthesisTaskOutput.isa = (o) => smithy_client_1.isa(o, "GetSpeechSynthesisTaskOutput");
})(GetSpeechSynthesisTaskOutput = exports.GetSpeechSynthesisTaskOutput || (exports.GetSpeechSynthesisTaskOutput = {}));
var InvalidLexiconException;
(function (InvalidLexiconException) {
    InvalidLexiconException.isa = (o) => smithy_client_1.isa(o, "InvalidLexiconException");
})(InvalidLexiconException = exports.InvalidLexiconException || (exports.InvalidLexiconException = {}));
var InvalidNextTokenException;
(function (InvalidNextTokenException) {
    InvalidNextTokenException.isa = (o) => smithy_client_1.isa(o, "InvalidNextTokenException");
})(InvalidNextTokenException = exports.InvalidNextTokenException || (exports.InvalidNextTokenException = {}));
var InvalidS3BucketException;
(function (InvalidS3BucketException) {
    InvalidS3BucketException.isa = (o) => smithy_client_1.isa(o, "InvalidS3BucketException");
})(InvalidS3BucketException = exports.InvalidS3BucketException || (exports.InvalidS3BucketException = {}));
var InvalidS3KeyException;
(function (InvalidS3KeyException) {
    InvalidS3KeyException.isa = (o) => smithy_client_1.isa(o, "InvalidS3KeyException");
})(InvalidS3KeyException = exports.InvalidS3KeyException || (exports.InvalidS3KeyException = {}));
var InvalidSampleRateException;
(function (InvalidSampleRateException) {
    InvalidSampleRateException.isa = (o) => smithy_client_1.isa(o, "InvalidSampleRateException");
})(InvalidSampleRateException = exports.InvalidSampleRateException || (exports.InvalidSampleRateException = {}));
var InvalidSnsTopicArnException;
(function (InvalidSnsTopicArnException) {
    InvalidSnsTopicArnException.isa = (o) => smithy_client_1.isa(o, "InvalidSnsTopicArnException");
})(InvalidSnsTopicArnException = exports.InvalidSnsTopicArnException || (exports.InvalidSnsTopicArnException = {}));
var InvalidSsmlException;
(function (InvalidSsmlException) {
    InvalidSsmlException.isa = (o) => smithy_client_1.isa(o, "InvalidSsmlException");
})(InvalidSsmlException = exports.InvalidSsmlException || (exports.InvalidSsmlException = {}));
var InvalidTaskIdException;
(function (InvalidTaskIdException) {
    InvalidTaskIdException.isa = (o) => smithy_client_1.isa(o, "InvalidTaskIdException");
})(InvalidTaskIdException = exports.InvalidTaskIdException || (exports.InvalidTaskIdException = {}));
var LanguageNotSupportedException;
(function (LanguageNotSupportedException) {
    LanguageNotSupportedException.isa = (o) => smithy_client_1.isa(o, "LanguageNotSupportedException");
})(LanguageNotSupportedException = exports.LanguageNotSupportedException || (exports.LanguageNotSupportedException = {}));
var Lexicon;
(function (Lexicon) {
    Lexicon.isa = (o) => smithy_client_1.isa(o, "Lexicon");
})(Lexicon = exports.Lexicon || (exports.Lexicon = {}));
var LexiconAttributes;
(function (LexiconAttributes) {
    LexiconAttributes.isa = (o) => smithy_client_1.isa(o, "LexiconAttributes");
})(LexiconAttributes = exports.LexiconAttributes || (exports.LexiconAttributes = {}));
var LexiconDescription;
(function (LexiconDescription) {
    LexiconDescription.isa = (o) => smithy_client_1.isa(o, "LexiconDescription");
})(LexiconDescription = exports.LexiconDescription || (exports.LexiconDescription = {}));
var LexiconNotFoundException;
(function (LexiconNotFoundException) {
    LexiconNotFoundException.isa = (o) => smithy_client_1.isa(o, "LexiconNotFoundException");
})(LexiconNotFoundException = exports.LexiconNotFoundException || (exports.LexiconNotFoundException = {}));
var LexiconSizeExceededException;
(function (LexiconSizeExceededException) {
    LexiconSizeExceededException.isa = (o) => smithy_client_1.isa(o, "LexiconSizeExceededException");
})(LexiconSizeExceededException = exports.LexiconSizeExceededException || (exports.LexiconSizeExceededException = {}));
var ListLexiconsInput;
(function (ListLexiconsInput) {
    ListLexiconsInput.isa = (o) => smithy_client_1.isa(o, "ListLexiconsInput");
})(ListLexiconsInput = exports.ListLexiconsInput || (exports.ListLexiconsInput = {}));
var ListLexiconsOutput;
(function (ListLexiconsOutput) {
    ListLexiconsOutput.isa = (o) => smithy_client_1.isa(o, "ListLexiconsOutput");
})(ListLexiconsOutput = exports.ListLexiconsOutput || (exports.ListLexiconsOutput = {}));
var ListSpeechSynthesisTasksInput;
(function (ListSpeechSynthesisTasksInput) {
    ListSpeechSynthesisTasksInput.isa = (o) => smithy_client_1.isa(o, "ListSpeechSynthesisTasksInput");
})(ListSpeechSynthesisTasksInput = exports.ListSpeechSynthesisTasksInput || (exports.ListSpeechSynthesisTasksInput = {}));
var ListSpeechSynthesisTasksOutput;
(function (ListSpeechSynthesisTasksOutput) {
    ListSpeechSynthesisTasksOutput.isa = (o) => smithy_client_1.isa(o, "ListSpeechSynthesisTasksOutput");
})(ListSpeechSynthesisTasksOutput = exports.ListSpeechSynthesisTasksOutput || (exports.ListSpeechSynthesisTasksOutput = {}));
var MarksNotSupportedForFormatException;
(function (MarksNotSupportedForFormatException) {
    MarksNotSupportedForFormatException.isa = (o) => smithy_client_1.isa(o, "MarksNotSupportedForFormatException");
})(MarksNotSupportedForFormatException = exports.MarksNotSupportedForFormatException || (exports.MarksNotSupportedForFormatException = {}));
var MaxLexemeLengthExceededException;
(function (MaxLexemeLengthExceededException) {
    MaxLexemeLengthExceededException.isa = (o) => smithy_client_1.isa(o, "MaxLexemeLengthExceededException");
})(MaxLexemeLengthExceededException = exports.MaxLexemeLengthExceededException || (exports.MaxLexemeLengthExceededException = {}));
var MaxLexiconsNumberExceededException;
(function (MaxLexiconsNumberExceededException) {
    MaxLexiconsNumberExceededException.isa = (o) => smithy_client_1.isa(o, "MaxLexiconsNumberExceededException");
})(MaxLexiconsNumberExceededException = exports.MaxLexiconsNumberExceededException || (exports.MaxLexiconsNumberExceededException = {}));
var OutputFormat;
(function (OutputFormat) {
    OutputFormat["JSON"] = "json";
    OutputFormat["MP3"] = "mp3";
    OutputFormat["OGG_VORBIS"] = "ogg_vorbis";
    OutputFormat["PCM"] = "pcm";
})(OutputFormat = exports.OutputFormat || (exports.OutputFormat = {}));
var PutLexiconInput;
(function (PutLexiconInput) {
    PutLexiconInput.isa = (o) => smithy_client_1.isa(o, "PutLexiconInput");
})(PutLexiconInput = exports.PutLexiconInput || (exports.PutLexiconInput = {}));
var PutLexiconOutput;
(function (PutLexiconOutput) {
    PutLexiconOutput.isa = (o) => smithy_client_1.isa(o, "PutLexiconOutput");
})(PutLexiconOutput = exports.PutLexiconOutput || (exports.PutLexiconOutput = {}));
var ServiceFailureException;
(function (ServiceFailureException) {
    ServiceFailureException.isa = (o) => smithy_client_1.isa(o, "ServiceFailureException");
})(ServiceFailureException = exports.ServiceFailureException || (exports.ServiceFailureException = {}));
var SpeechMarkType;
(function (SpeechMarkType) {
    SpeechMarkType["SENTENCE"] = "sentence";
    SpeechMarkType["SSML"] = "ssml";
    SpeechMarkType["VISEME"] = "viseme";
    SpeechMarkType["WORD"] = "word";
})(SpeechMarkType = exports.SpeechMarkType || (exports.SpeechMarkType = {}));
var SsmlMarksNotSupportedForTextTypeException;
(function (SsmlMarksNotSupportedForTextTypeException) {
    SsmlMarksNotSupportedForTextTypeException.isa = (o) => smithy_client_1.isa(o, "SsmlMarksNotSupportedForTextTypeException");
})(SsmlMarksNotSupportedForTextTypeException = exports.SsmlMarksNotSupportedForTextTypeException || (exports.SsmlMarksNotSupportedForTextTypeException = {}));
var StartSpeechSynthesisTaskInput;
(function (StartSpeechSynthesisTaskInput) {
    StartSpeechSynthesisTaskInput.isa = (o) => smithy_client_1.isa(o, "StartSpeechSynthesisTaskInput");
})(StartSpeechSynthesisTaskInput = exports.StartSpeechSynthesisTaskInput || (exports.StartSpeechSynthesisTaskInput = {}));
var StartSpeechSynthesisTaskOutput;
(function (StartSpeechSynthesisTaskOutput) {
    StartSpeechSynthesisTaskOutput.isa = (o) => smithy_client_1.isa(o, "StartSpeechSynthesisTaskOutput");
})(StartSpeechSynthesisTaskOutput = exports.StartSpeechSynthesisTaskOutput || (exports.StartSpeechSynthesisTaskOutput = {}));
var SynthesisTask;
(function (SynthesisTask) {
    SynthesisTask.isa = (o) => smithy_client_1.isa(o, "SynthesisTask");
})(SynthesisTask = exports.SynthesisTask || (exports.SynthesisTask = {}));
var SynthesisTaskNotFoundException;
(function (SynthesisTaskNotFoundException) {
    SynthesisTaskNotFoundException.isa = (o) => smithy_client_1.isa(o, "SynthesisTaskNotFoundException");
})(SynthesisTaskNotFoundException = exports.SynthesisTaskNotFoundException || (exports.SynthesisTaskNotFoundException = {}));
var SynthesizeSpeechInput;
(function (SynthesizeSpeechInput) {
    SynthesizeSpeechInput.isa = (o) => smithy_client_1.isa(o, "SynthesizeSpeechInput");
})(SynthesizeSpeechInput = exports.SynthesizeSpeechInput || (exports.SynthesizeSpeechInput = {}));
var SynthesizeSpeechOutput;
(function (SynthesizeSpeechOutput) {
    SynthesizeSpeechOutput.isa = (o) => smithy_client_1.isa(o, "SynthesizeSpeechOutput");
})(SynthesizeSpeechOutput = exports.SynthesizeSpeechOutput || (exports.SynthesizeSpeechOutput = {}));
var TaskStatus;
(function (TaskStatus) {
    TaskStatus["COMPLETED"] = "completed";
    TaskStatus["FAILED"] = "failed";
    TaskStatus["IN_PROGRESS"] = "inProgress";
    TaskStatus["SCHEDULED"] = "scheduled";
})(TaskStatus = exports.TaskStatus || (exports.TaskStatus = {}));
var TextLengthExceededException;
(function (TextLengthExceededException) {
    TextLengthExceededException.isa = (o) => smithy_client_1.isa(o, "TextLengthExceededException");
})(TextLengthExceededException = exports.TextLengthExceededException || (exports.TextLengthExceededException = {}));
var TextType;
(function (TextType) {
    TextType["SSML"] = "ssml";
    TextType["TEXT"] = "text";
})(TextType = exports.TextType || (exports.TextType = {}));
var UnsupportedPlsAlphabetException;
(function (UnsupportedPlsAlphabetException) {
    UnsupportedPlsAlphabetException.isa = (o) => smithy_client_1.isa(o, "UnsupportedPlsAlphabetException");
})(UnsupportedPlsAlphabetException = exports.UnsupportedPlsAlphabetException || (exports.UnsupportedPlsAlphabetException = {}));
var UnsupportedPlsLanguageException;
(function (UnsupportedPlsLanguageException) {
    UnsupportedPlsLanguageException.isa = (o) => smithy_client_1.isa(o, "UnsupportedPlsLanguageException");
})(UnsupportedPlsLanguageException = exports.UnsupportedPlsLanguageException || (exports.UnsupportedPlsLanguageException = {}));
var Voice;
(function (Voice) {
    Voice.isa = (o) => smithy_client_1.isa(o, "Voice");
})(Voice = exports.Voice || (exports.Voice = {}));
